compiler/rustc_codegen_llvm/src/debuginfo/mod.rs

```diff
@@ -97,7 +97,11 @@ impl<'ll, 'tcx> CodegenUnitDebugContext<'ll, 'tcx> {
                 // Android has the same issue (#22398)
                 llvm::add_module_flag_u32(
                     self.llmod,
-                    llvm::ModuleFlagMergeBehavior::Warning,
+                    // In the case where multiple CGUs with different dwarf version
+                    // values are being merged together, such as with cross-crate
+                    // LTO, then we want to use the highest version of dwarf
+                    // we can. This matches Clang's behavior as well.
+                    llvm::ModuleFlagMergeBehavior::Max,
```

Comment:
Question: do we have any public-facing docs for this behavior (e.g. next to `-Z dwarf-version` or sth?)

Reply:
I'm struggling to think of an example where this would cause a problem for users. The most obvious thing would be if your debugger (or other tool) does not support DWARF 5 (and thus errors out or whatever) but in that case the actual issue is that you've used `-Zdwarf-version=5` somewhere in your crate graph. The CGU merging behavior is only incidental at that point.

Maybe it would make sense to note something in `-Clto` but even then this seems to me like implementation minutia that's probably not suitable for user-facing documentation. Do you feel differently? 🙂

Reply:
Nah. I don't think it's worth it tbh. Thanks for the elaboration.